src/Funds/ServiceProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace GiveTestData\Funds;
+
+use WP_CLI;
+use GiveTestData\TestData\Helpers\SettingsPage;
+use GiveTestData\Funds\AdminSettings\FundsSettings;
+use Give\ServiceProviders\ServiceProvider as GiveServiceProvider;
+
+	/**
+ * Class ServiceProvider
+ * @package GiveTestData\Funds
+ */
+class ServiceProvider implements GiveServiceProvider {
+	/**
+	 * @inheritDoc
+	 */
+	public function register() {
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function boot() {
+		// Add CLI commands
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::add_command( 'give test-funds', give()->make( FundCommand::class ) );
+		}
+
+		/**
+		 * Inject Fund ID into revenue data
+		 */
+		add_filter(
+			'give-test-data-revenue-definition',
+			function ( $args ) {
+				$args['fund_id'] = ( give()->make( RandomFundProvider::class ) )();
```

Comment:
Would it be possible to assign the Random Fund inside of the Fund Factory, so that the logic is encapsulated and not depending on a filter?

I try to avoid required filters and they are not easily discoverable, thus easily overlooked.

Reply:
a8b95a38